document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresFilterTypeExpressionVisitor.java

```diff
@@ -43,19 +45,23 @@ public String visit(final RelationalExpression expression) {
     SelectTypeExpression rhs = expression.getRhs();
 
     // Only an identifier LHS and a constant RHS is supported as of now.
-    PostgresSelectTypeExpressionVisitor lhsVisitor = new PostgresIdentifierExpressionVisitor();
     PostgresSelectTypeExpressionVisitor rhsVisitor = new PostgresConstantExpressionVisitor();
-
-    String fieldName = lhs.accept(lhsVisitor);
     Object value = rhs.accept(rhsVisitor);
+    PostgresIdentifierExpressionVisitor identifierVisitor =
+        new PostgresIdentifierExpressionVisitor();
+    PostgresSelectTypeExpressionVisitor lhsVisitor =
+        new PostgresDataAccessorIdentifierExpressionVisitor(postgresQueryParser, getType(value));
```

Comment:
I think we need `PostgresFunctionExpressionVisitor` to support it, right?

Reply:
Yes. This will come in a follow-up PR along with unit & integration tests.